### PR TITLE
Improve mathematical efficiency of AABBs

### PR DESCRIPTION
**Pull request by ewpratten**
This PR improves AABB math, and makes things more extensible for the future. #144

### CODE REVIEW

File: lib5k/src/main/java/io/github/frc5024/lib5k/utils/math/average/MovingAverage.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        }
+
+        // Accumulate T
+        @SuppressWarnings("unchecked")
+        T lastValue = (T) values[0];
+
+        for (Object o : values) {
+            @SuppressWarnings("unchecked")
+            T value = (T) o;
+            if (value != null && value != lastValue) {
```
Comment (wm-c) [inline lib5k/src/main/java/io/github/frc5024/lib5k/utils/math/average/MovingAverage.java:93]: Couldn't this cause problems if the sum of the first 2 elements is equal to the 3rd ie 5, 3, 8 or 5, 5, 10

**Review by wm-c — CHANGES_REQUESTED** [review]
Interesting way of getting around the multiple inheritance problem

File: lib5k/src/main/java/io/github/frc5024/lib5k/utils/math/average/MovingAverage.java (hunk 1 above)
Comment (ewpratten) [reply lib5k/src/main/java/io/github/frc5024/lib5k/utils/math/average/MovingAverage.java:93]: nope. im using `!=` not `!(value.equals(lastValue))`. The first checks if they are not references to the same object. The second checks if they are different values

File: lib5k/src/main/java/io/github/frc5024/lib5k/utils/MathWrapper.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+public class MathWrapper<T> {
+
+    // Operations
+    private final DualOperation<T> additionOperation;
+    private final DualOperation<T> subtractionOperation;
+    private final TwoTypeDualOperation<T, Double, T> multiplicationOperation;
+    private final TwoTypeDualOperation<T, Double, T> divisionOperation;
+
+    /**
+     * Create a MethWrapper
```
Comment (ewpratten) [inline lib5k/src/main/java/io/github/frc5024/lib5k/utils/MathWrapper.java:21]: I dont want to start up my IDE to fix this

**Pull request merged**